pkg/trace/otel.go

```diff
@@ -92,24 +92,42 @@ func (t *otelTracer) initTracer(ctx context.Context, serviceName string) error {
 	mp := noop.NewMeterProvider()
 	otel.SetMeterProvider(mp)
 
-	key, err := t.Otel.APIKey.Data(ctx)
-	if err != nil {
-		log.Error(ctx, "Unable to fetch otel API key", events.NewErrorInfo(err))
-	}
+	var exp *otlptrace.Exporter
 
-	headers := map[string]string{
-		"x-honeycomb-team":    strings.TrimSpace(string(key)),
-		"x-honeycomb-dataset": t.Otel.Dataset,
-	}
+	if t.Otel.CollectorEndpoint == "" {
+		key, err := t.Otel.APIKey.Data(ctx)
+		if err != nil {
+			log.Error(ctx, "Unable to fetch otel API key", events.NewErrorInfo(err))
+		}
 
-	client := otlptracegrpc.NewClient(
-		otlptracegrpc.WithEndpoint(t.Otel.Endpoint+":443"),
-		otlptracegrpc.WithHeaders(headers),
-	)
+		headers := map[string]string{
+			"x-honeycomb-team":    strings.TrimSpace(string(key)),
+			"x-honeycomb-dataset": t.Otel.Dataset,
+		}
 
-	exp, err := otlptrace.New(ctx, client)
-	if err != nil {
-		log.Error(ctx, "Unable to start trace exporter", events.NewErrorInfo(err))
+		client := otlptracegrpc.NewClient(
```

Comment:
I wonder if it might be cleaner to take the client construction out of here?

We could have something separate that builds the `otlptracegrpc.NewClient` and/or its options.  We could even have two, one for `Endpoint` vs. `CollectorEndpoint`, if that works out better.

Very optional, though.

Reply:
I think that makes sense. I'm not tied to the if/else here because I think it's cluttered.

Reply:
I've attempted to clean this up.